Libraries/Storage/AsyncStorage.js

```diff
@@ -406,86 +355,149 @@ var AsyncStorage = {
     });
   },
 
-  /**
-   * Batch operation to merge in existing and new values for a given set of
-   * keys. This assumes that the values are stringified JSON. Returns a
-   * `Promise` object.
-   *
-   * **NOTE**: This is not supported by all native implementations.
-   *
-   * @param keyValuePairs Array of key-value array for the items to merge.
-   * @param callback Function that will be called with an array of any
-   *    key-specific errors found.
-   * @returns A `Promise` object.
-   *
-   * @example <caption>Example</caption>
-   * // first user, initial values
-   * let UID234_object = {
-   *  name: 'Chris',
-   *  age: 30,
-   *  traits: {hair: 'brown', eyes: 'brown'},
-   * };
-   *
-   * // first user, delta values
-   * let UID234_delta = {
-   *  age: 31,
-   *  traits: {eyes: 'blue', shoe_size: 10},
-   * };
-   *
-   * // second user, initial values
-   * let UID345_object = {
-   *  name: 'Marge',
-   *  age: 25,
-   *  traits: {hair: 'blonde', eyes: 'blue'},
-   * };
-   *
-   * // second user, delta values
-   * let UID345_delta = {
-   *  age: 26,
-   *  traits: {eyes: 'green', shoe_size: 6},
-   * };
-   *
-   * let multi_set_pairs   = [['UID234', JSON.stringify(UID234_object)], ['UID345', JSON.stringify(UID345_object)]]
-   * let multi_merge_pairs = [['UID234', JSON.stringify(UID234_delta)], ['UID345', JSON.stringify(UID345_delta)]]
-   *
-   * AsyncStorage.multiSet(multi_set_pairs, (err) => {
-   *   AsyncStorage.multiMerge(multi_merge_pairs, (err) => {
-   *     AsyncStorage.multiGet(['UID234','UID345'], (err, stores) => {
-   *       stores.map( (result, i, store) => {
-   *         let key = store[i][0];
-   *         let val = store[i][1];
-   *         console.log(key, val);
-   *       });
-   *     });
-   *   });
-   * });
-   *
-   * // Console log results:
-   * // => UID234 {"name":"Chris","age":31,"traits":{"shoe_size":10,"hair":"brown","eyes":"blue"}}
-   * // => UID345 {"name":"Marge","age":26,"traits":{"shoe_size":6,"hair":"blonde","eyes":"green"}}
-   */
-  multiMerge: function(
-    keyValuePairs: Array<Array<string>>,
-    callback?: ?(errors: ?Array<Error>) => void
-  ): Promise {
-    return new Promise((resolve, reject) => {
-      RCTAsyncStorage.multiMerge(keyValuePairs, function(errors) {
-        var error = convertErrors(errors);
-        callback && callback(error);
-        if (error) {
-          reject(error);
-        } else {
-          resolve(null);
-        }
-      });
-    });
-  },
+  setBackend: function (
+    backend: AsyncStorageBackend
+  ) {
+    RCTAsyncStorage = backend;
+    // Not all native implementations support merge.
+    if (backend.multiMerge) {
+      AsyncStorage.mergeItem = mergeItem;
+      AsyncStorage.multiMerge = multiMerge;
```

Comment:
@mkonicek same as above. I don't want to mix a bunch of different changes in one PR